client/src/featureform/enums.py

```diff
@@ -52,9 +54,52 @@ class SourceType(Enum):
     DF_TRANSFORMATION = "DF"
     SQL_TRANSFORMATION = "SQL"
 
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.value == other
+        if isinstance(other, SourceType):
+            return self.value == other.value
+        return False
+
 
 @typechecked
 @dataclass
 class FilePrefix(Enum):
     S3 = "s3://"
     S3A = "s3a://"
+
+
+class FileFormat(Enum):
+    CSV = "csv"
+    PARQUET = "parquet"
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, str):
```

Comment:
why do we need this equals? Can't we just leverage the enum equals?

Reply:
I wasn't aware the `Enum` derived classes have an `equals` method. I got hung up on `'csv'  == FileFormat.CSV.value` and when I looked for a way to avoid indexing into the enum, all of the SO articles described adding dunder methods for comparison operations. Seemed cleaner, but I'm happy to revert it.

Reply:
ohhhh I see what you're doing. okk i like thi

Reply:
FileFormat(str, Enum)

Reply:
I wouldn't inherit from str, seems like a super hack (always prefer composition over inheritence), and never seen it in python code before, I think this is fine as-is

Reply:
it's actually really common to do this. I saw it a ton at doordash. It's actually so common that python released StrEnum in 3.11